src/main/java/Controllers/RegistrationServlet.java

```diff
@@ -0,0 +1,49 @@
+package Controllers;
+
+import Models.User.DuplicateEmailPresentException;
+import Models.User.User;
+import Models.User.UserDao;
+import Utils.Tuple;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.security.NoSuchAlgorithmException;
+import java.security.spec.InvalidKeySpecException;
+
+import static Utils.Passwords.createSaltAndHash;
+
+@WebServlet(name = "RegistrationServlet")
+public class RegistrationServlet extends HttpServlet {
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+
+        try {
+            Tuple<byte[], byte[]> saltAndHash = createSaltAndHash(request.getParameter("submitted-password"));
```

Comment:
The benefit of `Tuple` is that we can return two separate values from a single function. Is that correct?

Reply:
Yeah, just lets you return two individual types rather than returning an array or arraylist of objects or some other gross hack - just lets me be a bit more descriptive of what data a function is returning